### ISSUE TITLE
Set Authorship colors default to false

### ISSUE THREAD

**Issue by pozzo-balbi** [ether/etherpad-lite#5563]
* * *

name: Option to set Authorship colors default to false
about: 
title: ''
labels: enhancement
assignees: 

* * *

**Is your feature request related to a problem? Please describe.**
![Screenshot at 2022-07-13 03-47-43](https://user-images.githubusercontent.com/3755138/178628386-454f6e93-da4c-4b5d-9529-c4e744ef97c6.png)
When you open a new pad you only have the options to disable colors altogether or have Authorship colors enabled. The problem is that you cannot activate authorship colors, once it is disabled, so disabling is not really a solution. And having authorship colors enabled by default has readability problems to new users. Sure, they can manually disable authorship colors, but if I need to explain how it is done, then maybe changing the default is the easier way to go about.

**Describe the solution you'd like**
In settings.json have an option
```
  "padOptions": {
    "noColors":         false,
    **"showAuthorColors": false,**
    "showControls":     true,
    "showChat":         true,
```

**Describe alternatives you've considered**
none were successful

**Additional context**
I found that this has already been requested here: https://groups.google.com/g/etherpad-lite-dev/c/bsIWRm73_Eg/m/7pJkKph-MOUJ . I found that changing the Authorship color box sets two values (showAuthorshipColors, showAuthorColors) in a cookie. 

**Plugin?**
Might this feature be better suited to being a plugin?
No, it should be in the settings.

**Comment by klardotsh** [comment]
While moving to https://www.npmjs.com/package/ep_author_neat2 made for a significantly better UX for my users, the default being `ON` for such a visually cluttered feature seems pretty horrible to me, and something I should be able to disable at a server level by default (unless the client side cookies have explicitly opted in). Huge +1 to this nearly two year old issue.

**Comment by robinmetral** [comment]
Agreed, in my opinion this isn't a sensible default as it distracts from the main goal of a document—the contents itself. Authorship is meta information that, while useful, should be toggled _on_ if needed.

I can imagine that there's some disagreement about this though, so at minimum an option to turn this off by default for given pads or server instances would be nice.

**Comment by robertgarrigos** [comment]
Just got to this page thinking there was no such option, but there is:

```
"padOptions": {
    "noColors":         true
}
}}}

**Comment by robinmetral** [comment]
@robertgarrigos does this actually hide authorship colors, or is it other kinds of colors? The settings snippet shared by OP already shows the `noColors` option.

(Disclosure: I haven't tested this, I don't run an etherpad server and I'm personally more interested in the option to disable authorship colors when creating a pad as a user of a shared instance.)

**Comment by robertgarrigos** [comment]
@robinmetral this is a server setting to hide authorship colors, indeed. I see, I misunderstood the issue. Sorry.

**Comment by JohnMcLear** [comment]
What does noColors do if not disable authorship colours?!

**Comment by robinmetral** [comment]
I think we're discussing two different things:

1. a server setting to disable authorship colors
2. a pad setting to disable authorship colors by default

For the server setting (1), it looks like it is in fact the `noColors` flag that was mentioned [above](https://github.com/ether/etherpad-lite/issues/5563#issuecomment-2675193016). (Haven't tested it but the code looks clear enough.)

Anyone with an Etherpad server wanting to disable authorship colors can set the flag to `true`:

https://github.com/ether/etherpad-lite/blob/555ccbe7b30e662d55cb4c397871f38688564e37/settings.json.docker#L275

For the default pad setting (2), as far as I can tell the option doesn't exist in the UI. Anyone visiting a pad would have to disable authorship colors themselves using "Hide authorship colors" in the settings.

A workaround is to share a pad link using the `noColors` query parameter set to `true`:

https://github.com/ether/etherpad-lite/blob/555ccbe7b30e662d55cb4c397871f38688564e37/src/static/js/pad.ts#L64-L71

Try it out (pad deletes after 30 days):

- `noColors=false` (default): https://yopad.eu/p/test_colors?noColors=false
- `noColors=true`: https://yopad.eu/p/test_colors?noColors=true

I still think that a UI option would be superior to a query parameter.

**Comment by JohnMcLear** [comment]
Afaik there should be testing in for noColors on the server but I'll double check that..  

1. I noticed if you set noColors on the pad in settings it doesn't show noColors
2. What's wrong with clicking Settings > Authorship colors.  Is the problem that it does not persists across pads???

**Comment by JohnMcLear** [comment]
I "think" I understood the problem ;)

**Comment by robinmetral** [comment]
> 2. What's wrong with clicking Settings > Authorship colors.  Is the problem that it does not persists across pads???

I'll try to clarify:

The issue is that a pad editor can't disable authorship colors for everyone else by default. Basically, Settings > Authorship colors is a device setting, not a pad setting.

Let's say I create a pad and want to share it with friends. I don't want them to be distracted by authorship colors, but as far as I know there's no way to disable them (except by sharing a URL with `?noColors=true`). Everyone would have to disable them on their own device.

To test this:

- Create a pad
- Write something, verify authorship colors are showing (they are on by default)
- Turn Settings > Authorship colors off
- Open the pad URL in a private tab (this is like sending the URL to another editor)
- Authorship colors are on—the setting is a device setting and not a pad setting

Don't get me wrong: the feature works as expected (the settings menu says "My view" so it's clearly meant to be a device setting), but I still think that a pad setting would be useful to turn off authorship colors by default for all editors. Hope it makes sense :)

**Comment by JohnMcLear** [comment]
@robinmetral That is desired behavior..  All settings in settings are **user** settings, not pad settings.

It sounds to me like you want to be the admin and change the settings on your own instance, that's the confusion here..  Etherpad can be deployed by you, not depending on some third party hosting :)  That, or, you can speak to whatever instance you use and ask them to change the setting in settings.json :)

**Comment by robinmetral** [comment]
@JohnMcLear I get it—I just don't see why there couldn't be a pad setting in addition to server and user settings. I don't need this badly enough to justify hosting my own instance, but that doesn't make it less of an annoyance for users of shared instances. In general I think that authorship colors should be off by default but that's another discussion.

Anyways, I'll be sharing pads using `?noColors=true` and that kind of solves the problem.

**Comment by JohnMcLear** [comment]
There is in develop.

The first author of the pad gets control and can force it.

🎁🔥
